hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -731,152 +722,21 @@ private void rollback(
     private PreHandleResult getCurrentPreHandleResult(
             @NonNull final ReadableStoreFactory storeFactory,
             @NonNull final NodeInfo creator,
-            @NonNull final ConsensusTransaction platformTxn,
-            @NonNull final VersionedConfiguration configuration)
-            throws PreCheckException {
+            @NonNull final ConsensusTransaction platformTxn) {
         final var metadata = platformTxn.getMetadata();
+        final PreHandleResult previousResult;
+        if (metadata instanceof PreHandleResult result) {
```

Comment:
We could probably add a comment here when will this result be null and when it is not 😇 .

Reply:
Ah true, true. 😇

Reply:
Added `logger.error()` in `else` block since metadata should _always_ be a `PreHandleResult`.